bot/exts/evergreen/status_codes.py

```diff
@@ -21,49 +22,28 @@ async def http_status_group(self, ctx: commands.Context) -> None:
 
     @http_status_group.command(name='cat')
     async def http_cat(self, ctx: commands.Context, code: int) -> None:
-        """Sends an embed with an image of a cat, portraying the status code."""
-        embed = discord.Embed(title=f'**Status: {code}**')
-        url = HTTP_CAT_URL.format(code=code)
-
-        try:
-            HTTPStatus(code)
-            async with self.bot.http_session.get(url, allow_redirects=False) as response:
-                if response.status != 404:
-                    embed.set_image(url=url)
-                else:
-                    raise NotImplementedError
-
-        except ValueError:
-            embed.set_footer(text='Inputted status code does not exist.')
-
-        except NotImplementedError:
-            embed.set_footer(text='Inputted status code is not implemented by http.cat yet.')
-
-        finally:
-            await ctx.send(embed=embed)
+        """Assemble Cat URL and build Embed."""
+        await self.build_embed(url=HTTP_CAT_URL.format(code=code), ctx=ctx, code=code)
 
     @http_status_group.command(name='dog')
     async def http_dog(self, ctx: commands.Context, code: int) -> None:
-        """Sends an embed with an image of a dog, portraying the status code."""
-        embed = discord.Embed(title=f'**Status: {code}**')
-        url = HTTP_DOG_URL.format(code=code)
-
-        try:
-            HTTPStatus(code)
-            async with self.bot.http_session.get(url, allow_redirects=False) as response:
-                if response.status != 302:
-                    embed.set_image(url=url)
-                else:
-                    raise NotImplementedError
-
-        except ValueError:
-            embed.set_footer(text='Inputted status code does not exist.')
-
-        except NotImplementedError:
-            embed.set_footer(text='Inputted status code is not implemented by httpstatusdogs.com yet.')
-
-        finally:
-            await ctx.send(embed=embed)
+        """Assemble Dog URL and build Embed."""
+        await self.build_embed(url=HTTP_DOG_URL.format(code=code), ctx=ctx, code=code)
+
+    async def build_embed(self, url: str, ctx: commands.Context, code: int) -> None:
+        """Attempt to build and dispatch embed. Append error message instead if something goes wrong."""
+        async with self.bot.http_session.get(url, allow_redirects=False) as response:
+            if 200 <= response.status <= 299:
+                await ctx.send(embed=discord.Embed(
+                    title=STATUS_TEMPLATE.format(code=code)).set_image(url=url))
+            elif 404 == response.status:
+                await ctx.send(embed=discord.Embed(
+                    title=ERR_404.format(code=code)).set_image(url=url)
+                )
+            else:
+                await ctx.send(embed=discord.Embed(
+                    title=STATUS_TEMPLATE.format(code=code), footer=ERR_UNKNOWN.format(code=code))
+                )
```

Comment:
The `embed=` part should be on a new line, and `footer` isn't a valid keyword argument for `Embed`. Use `Embed.set_footer` instead.
```suggestion
                await ctx.send(
                    embed=discord.Embed(
                        title=STATUS_TEMPLATE.format(code=code)
                    ).set_footer(text=ERR_UNKNOWN.format(code=code))
                )
```